src/universal-dapp.js

```diff
@@ -650,6 +666,7 @@ UniversalDApp.prototype.runTx = function (data, args, cb) {
 
   var tx;
   if (!self.vm) {
+    self.transactionDebugger.switchProvider('EXTERNAL');
```

Comment:
I think it is still better for now to make it depend on the debugger. There is a task to convert browser-solidity into an event-based environment where a provider switch would be an event everyone could listen on and act accordingly. It is better to have these calls in one place.